### ISSUE TITLE
Dropout1d (channel dropout for 1d data)

### ISSUE THREAD

**Issue by chanshing** [pytorch/pytorch#60906]
## 🚀 Feature

Something like `nn.Dropout2d` which does *channel* dropout (i.e. dropping an entire channel/feature map instead of bits across different channels) but for 1d data. Could be called `nn.Dropout1d`.

## Motivation

Those working with _multivariate_ time-series data (e.g. stock price of multiple companies, GPS data, activity tracker motion data, etc.) are lacking the ability to perform channel dropout. Right now, a workaround is to add a dummy axis to the input data and use `nn.Dropout2d`: 

```python
# x.shape is (B, C, N) (batch size, num channels, temporal length)
y = nn.Dropout2d()(x[..., None]).view(*x.shape)
```

cc @albanD @mruberry @jbschlosser

**Comment by jbschlosser** [comment]
Hey @chanshing, thanks for the suggestion! Unless I'm mistaken, contrary to what the docs say, `Dropout2d` supports entire channel dropout for 1D data without the need for a dummy axis:

```python
>>> torch.nn.Dropout2d()(torch.randn(2, 3, 4))
tensor([[[ 2.0013,  0.5137,  4.6231, -0.8030],
         [ 0.2068,  1.2131,  1.2506,  2.1023],
         [ 0.0000, -0.0000,  0.0000, -0.0000]],

        [[-2.2049, -4.3484,  0.4871,  1.2764],
         [-0.0000,  0.0000,  0.0000,  0.0000],
         [-1.0111, -0.5624,  0.7527, -0.0970]]])
```

Possible alternative may be to support a generic, properly-documented `Dropout` with configurable dims over which to dropout. See https://github.com/pytorch/pytorch/issues/46184 as well.

**Comment by chanshing** [comment]
Oh! I should have tried that. Thanks!
I think this could be clarified in the documentation https://pytorch.org/docs/stable/generated/torch.nn.Dropout2d.html
Currently, it seems to indicate it only works for 4D inputs.
